module/library/categories/helper.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * name : helper.js
+ * author : Aman
+ * created-date : 23-Jun-2020
+ * Description : Library categories helper functionality.
+ */
+
+// Dependencies 
+let kendraService = require(ROOT_PATH + "/generics/services/kendra");
+let sessionHelpers = require(ROOT_PATH+"/generics/helpers/sessions");
+
+/**
+    * libraryCategoriesHelper
+    * @class
+*/
+
+module.exports = class libraryCategoriesHelper {
+
+      /**
+   * Library categories
+   * @method
+   * @name categoryDocuments
+   * @param {Object} [findQuery = "all"] - filtered data.
+   * @param {Array} [fields = "all"] - projected data.
+   * @param {Array} [skipFields = "none"] - fields to skip.
+   * @returns {Array} - Library categories data.
+   */
+
+  static categoryDocuments(
+    findQuery = "all", 
+    fields = "all",
+    skipFields = "none"
+  ) {
+      return new Promise(async (resolve, reject) => {
+        
+        try {
+          
+          let queryObject = {};
+
+          if (findQuery != "all") {
+              queryObject = findQuery;
+          }
+
+          let projection = {};
+
+          if (fields != "all") {
+              fields.forEach(element => {
+                  projection[element] = 1;
+              });
+          }
+
+          if (skipFields != "none") {
+              skipFields.forEach(element => {
+                  projection[element] = 0;
+              });
+          }
+
+          let libraryCategoriesData = 
+          await database.models.libraryCategories.find(
+            queryObject, 
+            projection
+          ).lean();
+          
+          return resolve(libraryCategoriesData);
+
+      } catch (error) {
+          return reject({
+              status: error.status || httpStatusCode.internal_server_error.status,
+              message: error.message || httpStatusCode.internal_server_error.message,
+              errorObject: error
+          });
+        }
+      });
+    }
+
+    /**
+      * List of library categories.
+      * @method
+      * @name list
+      * @returns {Object} Library categories lists.
+     */
+
+    static list() {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let result = "";
+                let libraryData = sessionHelpers.get("libraryCategories");
+
+                if( libraryData && libraryData.length > 0 ) {
+
+                    result = libraryData;
+
+                } else {
+                    
+                    let libraryCategories = 
+                    await this.categoryDocuments(
+                        {
+                            status : messageConstants.common.ACTIVE_STATUS
+                        },
+                        [
+                            "externalId",
```

Comment:
@aman-tunerlabs - Give updated At in response, app can use it decide or not to refresh their local storage.